app/src/main/java/agency/five/codebase/android/movieapp/ui/home/HomeViewModel.kt

```diff
@@ -0,0 +1,133 @@
+    private val nowPlayingSelectedCategory: MutableState<MovieCategory> =
+        mutableStateOf(MovieCategory.NOW_PLAYING_TV)
+
+    private val _popularMoviesHomeViewState = MutableStateFlow(HomeMovieCategoryViewState.EMPTY)
+    val popularMoviesHomeViewState: StateFlow<HomeMovieCategoryViewState> =
+        _popularMoviesHomeViewState.asStateFlow()
+
+    private val _nowPlayingMoviesHomeViewState = MutableStateFlow(HomeMovieCategoryViewState.EMPTY)
+    val nowPlayingMoviesHomeViewState: StateFlow<HomeMovieCategoryViewState> =
+        _nowPlayingMoviesHomeViewState.asStateFlow()
+
+    private val _upcomingMoviesHomeViewState = MutableStateFlow(HomeMovieCategoryViewState.EMPTY)
+    val upcomingMoviesHomeViewState: StateFlow<HomeMovieCategoryViewState> =
+        _upcomingMoviesHomeViewState.asStateFlow()
+
+    init {
+        getPopularMovies()
+        getUpcomingMovies()
+        getNowPlayingMovies()
+    }
+
+    private fun getPopularMovies() {
+        viewModelScope.launch {
+            movieRepository.popularMovies(MovieCategory.POPULAR_STREAMING).collect {
+                _popularMoviesHomeViewState.value = homeScreenMapper.toHomeMovieCategoryViewState(
+                    movieCategories = popularCategories,
+                    selectedMovieCategory = popularSelectedCategory.value,
+                    movies = it
+                )
+            }
+        }
+    }
+
+    private fun getUpcomingMovies() {
+        viewModelScope.launch {
+            movieRepository.upcomingMovies(MovieCategory.UPCOMING_TODAY).collect {
+                _upcomingMoviesHomeViewState.value = homeScreenMapper.toHomeMovieCategoryViewState(
+                    movieCategories = upcomingCategories,
+                    selectedMovieCategory = upcomingSelectedCategory.value,
+                    movies = it
+                )
+            }
+        }
+    }
+
+    private fun getNowPlayingMovies() {
+        viewModelScope.launch {
+            movieRepository.nowPlayingMovies(MovieCategory.NOW_PLAYING_MOVIES).collect {
+                _nowPlayingMoviesHomeViewState.value =
+                    homeScreenMapper.toHomeMovieCategoryViewState(
+                        movieCategories = nowPlayingCategories,
+                        selectedMovieCategory = nowPlayingSelectedCategory.value,
+                        movies = it
+                    )
+            }
+        }
+    }
+
+    fun toggleFavorite(id: Int) {
+        viewModelScope.launch {
+            movieRepository.toggleFavorite(id)
+        }
+    }
+
+    fun changeCategory(categoryId: Int) {
+        when (categoryId) {
+            MovieCategory.POPULAR_STREAMING.ordinal,
+            MovieCategory.POPULAR_FOR_RENT.ordinal,
+            MovieCategory.POPULAR_ON_TV.ordinal,
+            MovieCategory.POPULAR_IN_THEATRES.ordinal
+            -> {
+                popularSelectedCategory.value = MovieCategory.values()[categoryId]
+                getPopularMovies()
```

Comment:
this should be removes, same goes for `getNowPlayingMovies` and `getupcomingMovies` methods.